### ISSUE TITLE
Question : how to use typedefinition with global declarations  (as opposed to files declared as modules) ?

### ISSUE THREAD

**Issue by valeriob** [ReactiveX/rxjs#4374]
For instance what other npm libraries do is to publish the @types package (that i can't find if there is one) or use the convention to add a "types" property to package.json pointing at a folder with the definition.
Thanks

**Comment by kwonoj** [comment]
if you're referring umd, there isn't way. check https://github.com/ReactiveX/rxjs/issues/1481 for discussion - in short we are waiting upstream compiler support and do not plan to manually handcraft global definitions.

**Comment by valeriob** [comment]
I'm not talking about UMD but ESM, and it works : 
Add a line to index.d.ts "export as namespace rxjs;", it's supposed to support this exact scenario, like many other packages are doing.

**Comment by kwonoj** [comment]
for esm and any other non-umd build definitions, we don't support global. We do only provide per-module based type generated by compiler and will not make manual or handcraft modification for it.

**Comment by kwonoj** [comment]
Just clarify - any packages belong to `@types` is 3rd-party types, not direct generated by compiler from typescript source code and can't be 1:1 compared. We are open to support once compiler can support given scenarios if it does.

**Comment by valeriob** [comment]
> for esm and any other non-umd build definitions, we don't support global

Why would you make such a radical choice, given that the path to support this scenario is well covered, documented and just one like that does not break modules ?

**Comment by kwonoj** [comment]
1. If you are using ESM build, that's already per-file module and it is definitely not _globally_ exported. Type definition file do not carry both.

2. Even if type doesn't provide global definitions, you are not blocked by it, ts compiler allows user-level definition like `declare const xxx` with some type you'd like to assign, which creates global scope types for you.

What you described above is very unusual case, you mentioned you are using module system build and then you says you are not using module. For non-module build (umd), this is def. caveat and we'd like to improve but for module build there's no point manually handcraft types for this specific cases.

If you feel you are completely blocked, plz feel free to create new issue with details, specifically `github repo having mininal source code with your build setup` and we can find proper setup.

**Comment by valeriob** [comment]
ok thanks, i'll try to do that.

**Comment by cartant** [comment]
@valeriob You might find this useful/relevant: https://davidea.st/articles/typescript-2-9-import-types

Maybe. I've not had to mess with the scenario you've described.

**Comment by valeriob** [comment]
Thanks @cartant i'll look into it.
The scenario i'm describing is very simple, write an html page, npm install rxjs, write some typescript Index.ts that uses it, run tsc with with modules="none", import in that page rxjs file and the index.js file generated.
